qiskit_optimization/translators/utils.py

```diff
@@ -0,0 +1,51 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utilities for optimization model translators"""
+
+from typing import TYPE_CHECKING, Any
+
+from qiskit_optimization.exceptions import QiskitOptimizationError
+
+from .docplex_mp import DocplexMpTranslator
+from .gurobi import GurobiTranslator
+from .lp_file import LPFileTranslator
+
+if TYPE_CHECKING:
+    # pylint: disable=cyclic-import
+    from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+
+translators = [DocplexMpTranslator(), GurobiTranslator(), LPFileTranslator()]
+
+
+def _load_model(model: Any) -> "QuadraticProgram":
```

Comment:
@adekusar-drl suggested to hide `utils.*` from users. We thought `QP.load` is enough. https://github.com/Qiskit/qiskit-optimization/pull/122#discussion_r630091813

I agree to make `translators` private, `is_compatible` a class method. We can make `is_installed` a class method too.

Reply:
I tried to make `_translators = [DocplexMpTranslator, ..., ]`, but mypy fails as follows. We need perhaps a trick of mypy. I avoid it by instantiate objects.
```
mypy qiskit_optimization test tools
qiskit_optimization/translators/utils.py:46: error: "ABCMeta" has no attribute "is_installed"
qiskit_optimization/translators/utils.py:46: error: "ABCMeta" has no attribute "is_compatible"
```

Reply:
Maybe it needs to be typehinted as list of ModelTranslators

Reply:
Thanks. I tried `_translator_types: List[Type[QuadraticProgramTranslator]]` and it works.

Reply:
I'm thinking of moving the code of `_load_model` to `QP.load` because it is not so long. But, importing all translators in QP might be complicated. So, I may leave `_translator_types` in utils.py as a private list.